diff --git a/audio.c b/audio.c
index 5ae0e1e..863238f 100644
--- a/audio.c
+++ b/audio.c
@@ -883,7 +883,7 @@ static void find_working_driver (lists_t_strs *drivers, struct hw_funcs *funcs)
 {
 	int ix;
 
-	memset (funcs, 0, sizeof(funcs));
+	memset (funcs, 0, sizeof(*funcs));
 
 	for (ix = 0; ix < lists_strs_size (drivers); ix += 1) {
 		const char *name;
diff --git a/decoder_plugins/ffmpeg/ffmpeg.c b/decoder_plugins/ffmpeg/ffmpeg.c
index a85fd82..e8f0b91 100644
--- a/decoder_plugins/ffmpeg/ffmpeg.c
+++ b/decoder_plugins/ffmpeg/ffmpeg.c
@@ -301,6 +301,10 @@ static int locking_cb (void **mutex, enum AVLockOp op)
 		free (*mutex);
 		*mutex = NULL;
 		break;
+	default:
+		/* Catch any FFmpeg/LibAV API changes. */
+		assert (0);
+		result = 0;
 	}
 
 	return result;
