diff --git a/decoder_plugins/mp3/xing.c b/decoder_plugins/mp3/xing.c
index 743cc3d..68a01dc 100644
--- a/decoder_plugins/mp3/xing.c
+++ b/decoder_plugins/mp3/xing.c
@@ -13,8 +13,8 @@
  * GNU General Public License for more details.
  *
  * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  *
  * $Id: xing.c,v 1.5 2002/11/18 16:32:21 daper Exp $
  */
diff --git a/decoder_plugins/mp3/xing.h b/decoder_plugins/mp3/xing.h
index 8cb1c2a..1f26e09 100644
--- a/decoder_plugins/mp3/xing.h
+++ b/decoder_plugins/mp3/xing.h
@@ -13,8 +13,8 @@
  * GNU General Public License for more details.
  *
  * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  *
  * $Id: xing.h,v 1.3 2002/10/04 13:38:09 daper Exp $
  */
